tasks/21.txt

```diff
@@ -1,3 +1,15 @@
 Write a SQL statement to get the maximum purchase amount of all the orders.
 
-| Order Id | Maximum Purchase Amount, $ |
\ No newline at end of file
+| Order Id | Maximum Purchase Amount, $ |
+
+//task21
+  db.getCollection('order-details').aggregate([  
+    { $group:
+          {
+            _id: { OrderID: "$OrderID"}, 
+            Amount: {$max: { $sum : { $multiply: [ "$UnitPrice", "$Quantity" ] } }}
+          }
+      },
+     { $sort: { Amount: -1}},
+     { $limit : 1 }, 
+  ])
```

Comment:
{$max: { $sum : { $multiply: [ "$UnitPrice", "$Quantity" ] } }}

We can't use 2 aggregation function in one $group ($max + $sum in your case). That doesn't have a sense.
What we have actually here
{ $multiply: [ "$UnitPrice", "$Quantity" ] } - projection 
{ $sum : - group aggregation, will be just ignored in your solution
{ $max: - group aggregation